### ISSUE TITLE
Optimising build time

### ISSUE THREAD

**Issue by dragarcia** [supabase/postgres#13]
With the introduction of plv8, build times for new images have gone up to unfavourable timings. Biggest factor thus far is RAM allocated to the image upon building.

# Digital Ocean
- Increasing RAM at build time is not ideal as Digital Ocean plans are tagged to both RAM and disk space.
- Going for a plan with high RAM equates to higher disk space as well.
- We would be unable to scale down the image afterwards

## Possible Solutions
- ~~Build locally then deploy. (Perhaps with something like Vagrant)~~
- ~~Build on top of an existing image with v8 already installed.~~
- Build on top of previous version.

# AWS
- It is possible to increase RAM during build time and scale it accordingly in both directions for instances that would be produced from the resultant AMI.

# Docker
- Not that of a worry as building it is done locally first then eventually deployed on to Docker Hub.

**Comment by dragarcia** [comment]
**tl;dr** - We build on top of an existing version of Supabase Postgres (We take the one from the marketplace) instead of from scratch and use [Ansible tags](https://docs.ansible.com/ansible/latest/user_guide/playbooks_tags.html) to run specific tasks instead of everything. A guide on how to build on top can be found [here](https://github.com/supabase/postgres/wiki/Digital-Ocean#build-on-top-of-an-existing-version).

**Comment by dragarcia** [comment]
Will close this issue once I create the same configuration for AWS (now that we have Supabase Marketplace available in their marketplace).
